### DISCUSSION TITLE
Draw the line between two points in the space with given x and y values?

### DISCUSSION THREAD

**Discussion by VIJAYAYERUVA** [JasonKessler/scattertext#78] [Q&A] [answered]
Dear Jason,

Is there any way that I can draw the line between two points in the space with given x and y values?
Is there any existing method that will take a set of X and set Y of values to draw a line in the space (not the horizontal or vertical lines)?

The method [produce_scattertext_explorer](https://github.com/JasonKessler/scattertext/blob/6779d045351c03c24ee6aa6bca09a78bc461f422/scattertext/__init__.py#L123), will take X, Y values, and plot horizontal or vertical lines. However, I want to plot in the space with multiple X and Y values

Any help is appreciated. 

![scattertext](https://user-images.githubusercontent.com/25230005/102038872-009ef580-3d8e-11eb-83d9-c02b17dfdc5f.png)

**Comment by doraithodla** [comment]
Vijaya,

Will this help?  We use this in ML as a visual aid for prediction.

https://seaborn.pydata.org/tutorial/regression.html

All the libraries are open source.

Dorai
LinkedIn Profile <https://www.linkedin.com/in/doraithodla/>
@dorait <https://twitter.com/dorait>

**Reply by VIJAYAYERUVA** [reply]
Thank you, Dorai, for the reply. I have tried plotting with these methods as below. But I could not use the existing scattertext space. Because of that, the lines are inconsistent. I want to pass X and Y values existing methods `produce_scattertext_explorer` like the one above.

![Sca](https://user-images.githubusercontent.com/25230005/102040149-87a19d00-3d91-11eb-8eee-219752dad1c5.png)

**Comment by JasonKessler** [comment, marked as answer]
Hi @VIJAYAYERUVA ,

Right now there's no way to do this in Scattertext.

However, you can follow the template of https://github.com/JasonKessler/scattertext/commit/e1a9950d54d8da3c202fb6f0dc1f275092b35ae5, the commit that added a diagonal line from the lower-left to the upper-right corner of the plot.  It includes everything from adding parameter to produce_scattertext_explorer to the actual change in the JS file.

I encourage you to add the functionality which meets your needs, and contribute it back to the community.

Jason

**Reply by VIJAYAYERUVA** [reply]
Thank you for your response. I will do that.

**Reply by VIJAYAYERUVA** [reply]
I can draw the lines. Thanks for your inputs.

**Comment by VIJAYAYERUVA** [comment]
@JasonKessler, Is there any update to vertical_lines in `produce_scattertext_explorer`?
Previously, I am able to draw vertical lines. Now Vertical Lines are not visible even with `show_axes_and_cross_hairs=True`. 

Sample code

```
html = st.produce_scattertext_explorer(
    corpus,
    category='male',
    category_name='Male',
    not_category_name='Female',
    width_in_pixels=1000,
    height_in_pixels=700,
    show_axes_and_cross_hairs=True,
    vertical_lines = [0.2, 0.3, 0.7,0.8])

```

**Reply by JasonKessler** [reply]
Could you post a reproducible example of what's happening?

**Reply by VIJAYAYERUVA** [reply]
I have the below code, and I expected the below image.

### Code
![Code](https://github.com/VIJAYAYERUVA/Images2Share/blob/main/scattertext2_code.png)

### Image
![Error](https://github.com/VIJAYAYERUVA/Images2Share/blob/main/scattertext2_error.png)

**Reply by JasonKessler** [reply]
Unfortunately, we can't run code from an image, nor can we access your dataset. 

Could you please post an example anyone can run, using, say a public data set like the Conventions dataset included in the package?

**Reply by VIJAYAYERUVA** [reply]
I have the below code, and I expected the below image.

### Code

```
import scattertext as st

df = st.SampleCorpora.ConventionData2012.get_data().assign(
    parse=lambda df: df.text.apply(st.whitespace_nlp_with_sentences)
)

corpus = st.CorpusFromParsedDocuments(
    df, category_col='party', parsed_col='parse'
).build().get_unigram_corpus().compact(st.AssociationCompactor(2000))

html = st.produce_scattertext_explorer(
    corpus,
    category='democrat', category_name='Democratic', not_category_name='Republican',
    minimum_term_frequency=0, pmi_threshold_coefficient=0,
    width_in_pixels=1000, metadata=corpus.get_df()['speaker'],
    transform=st.Scalers.dense_rank,
    show_axes_and_cross_hairs=True,
    vertical_lines = [0.2, 0.3, 0.7,0.8])
open('./demo_compact.html', 'w').write(html)

```

###  Actual Image
![Sca](https://github.com/VIJAYAYERUVA/Images2Share/blob/main/scattertext3_error.png)

### Expected Image
![Sca](https://github.com/VIJAYAYERUVA/Images2Share/blob/main/scattertext3_expected.png)

**Reply by JasonKessler** [reply]
Could you ensure you are on version 0.0.2.75 and try again?

**Reply by VIJAYAYERUVA** [reply]
Sure. I think I am on the latest version. I will check it with 0.0.2.75.

**Reply by VIJAYAYERUVA** [reply]
Hey, I am on `scattertext 0.0.2.74 ` version

**Reply by VIJAYAYERUVA** [reply]
It is working with `scattertext 0.0.2.75` version
